service/group0_state_broadcaster.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "group0_state_broadcaster.hh"
+#include "compaction/compaction_manager.hh"
+#include "replica/database.hh"
+
+namespace service {
+
+namespace {
+
+logging::logger slogger("group0_state_broadcaster");
+
+} // namespace
+
+void group0_state_broadcaster::start() {
+    SCYLLA_ASSERT(this_shard_id() == 0);
+
+    if (_stopped) {
+        return;
+    }
+
+    _done = make_ready_future<>();
+
+    _timer.set_callback([this] {
+        slogger.debug("Disseminating last state id ...");
+
+        _done = sys_ks()
+                    .get_last_group0_state_id()
+                    .then([this](const utils::UUID& state_id) {
+                        return gossiper().add_local_application_state(gms::application_state::STATE_ID,
+                                                                      gms::versioned_value::state_id(service::state_id(state_id)));
```

Comment:
This could be done on each group 0 state machine apply, and only the collection would be done periodically.

Although even the collection would be probably simpler if it was a gossiper subscription instead of periodically running loop (then we'd be reacting almost instantaneously too).

Reply:
Do we want to do it on each apply creating more gossip traffic?

Reply:
Done, moved now to be triggered upon `group0_state_machine::apply()`. Also checking the last state so that we don't advertise the same state id multiple time (although ... maybe it is not even possible to get the exact same id into the `group0_state_machine::apply()` twice?)